src/main/java/org/jenkinsci/plugins/gittagmessage/GitTagMessageExtension.java

```diff
@@ -25,14 +27,21 @@
 public class GitTagMessageExtension extends GitSCMExtension {
 
     private static final Logger LOGGER = Logger.getLogger(GitTagMessageExtension.class.getName());
+    private static final Pattern tagAndOffsetMatcher = Pattern.compile("(?<tag>.+)-[0-9]+-g[0-9A-Fa-f]{7}$");
+
+    private final boolean useMostRecentTag;
 
     @DataBoundConstructor
-    public GitTagMessageExtension() {
-        // No configuration options
+    public GitTagMessageExtension(boolean useMostRecentTag) {
+        this.useMostRecentTag = useMostRecentTag;
+    }
```

Comment:
Since this is an optional parameter, I would prefer to keep the constructor empty, and set `useMostRecentTag` via a setter method annotated with `@DataBoundSetter`.